### ISSUE TITLE
Controller inheritance

### ISSUE THREAD

**Issue by tijana-vujanac** [AlexTeixeira/Askmethat-Aspnet-JsonLocalizer#77]
Hi,

I have HomeController which inherits BaseController

Constructor of HomeController:
```
 public HomeController(IConfiguration configuration,
            **IJsonStringLocalizer<HomeController> localizer**) :
        base(configuration, localizer){
}
```

Class  BaseController:

 ```
 private readonly IConfiguration _configuration;
  protected  IJsonStringLocalizer<BaseController> _localizer;

        public BaseController(IConfiguration configuration,
            **IJsonStringLocalizer<BaseController> localizer**)
        {
            this.apiHandler = apiHandler;
            this._configuration = configuration;
            this._localizer = localizer;
        }
```
If we want to use it this way, we would actually need (I assume):

` public interface IJsonStringLocalizer<out T> : IJsonStringLocalizer
`
instead of

` public interface IJsonStringLocalizer<T> : IJsonStringLocalizer`

<**T**> -> <**out T**>

**Comment by AlexTeixeira** [comment]
Hi,

Sorry for the delay.

I add the changes in the last code changes and be available in version 3.0.0 preview 3

**Comment by tijana-vujanac** [comment]
Great! Thank you!
